### PR TITLE
feat: support "remove" of any array index

### PR DESCRIPTION
**Pull request by aleclarson**
WDYT?

(no tests yet)

### CODE REVIEW

File: src/patches.js [hunk 1]
```diff
@@ -128,14 +128,26 @@ export function applyPatches(draft, patches) {
...
+                        const index = Number(key)
+                        if (!Number.isNaN(index)) {
+                            if (index < 0 || index > base.length) {
+                                throw new Error(
+                                    "Invalid array patch: The given index is out of bounds"
+                                )
+                            }
+                            if (index === 0) {
+                                base.shift()
+                                break
```
Comment (mweststrate) [inline src/patches.js]: NIT: `if else` ... might make this code block slighly clearer? Extracting utility code might also be nice

**Comment by mweststrate** [comment]
Eh no it doesn't :-). Will accept PR though if you add some tests for the specific cases (the coverage report is a great way to verify that all of them are covered!)

File: src/patches.js (hunk 1 above)
Comment (aleclarson) [reply src/patches.js]: Probably, I'm just not a fan of a lonesome `break`.

You want `break`s removed and `else`s added? Or keep the `breaks`?

Comment (aleclarson) [reply src/patches.js]: Oh I suppose you mean to break this out into its own function?

**Comment by mweststrate** [comment]
Yes

Op za 13 okt. 2018 02:48 schreef Alec Larson <notifications@github.com>:

**Comment by aleclarson** [comment]
Closing this for now.

**Pull request closed without merging**